File: crates/instrument-core/src/probe_policy.rs
use serde::{Deserialize, Serialize};

/// How aggressively discovery probes instrument capabilities.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, Serialize, Deserialize)]
#[non_exhaustive]
pub enum ProbePolicy {
    /// Registry, VISA attributes, and *IDN? only — no capability queries.
    None,
    /// Benign read-only state queries (default); does not trigger measurements or change outputs.
    #[default]
    ReadOnly,
    /// Includes acquisition-triggering probes such as `:MEAS:VOLT:DC?`.
    Full,
}

File: dotnet/src/InstrumentComponents/Probe/ProbePolicy.cs
namespace InstrumentComponents.Probe;

/// <summary>How aggressively discovery probes instrument capabilities.</summary>
public enum ProbePolicy
{
    /// <summary>Registry, VISA attributes, and *IDN? only — no capability queries.</summary>
    None,
    /// <summary>Benign read-only state queries (default).</summary>
    ReadOnly,
    /// <summary>Includes acquisition-triggering probes such as :MEAS:VOLT:DC?.</summary>
    Full,
}
